### PR TITLE
[WIP] Fixing node BGP STs

### PR DESCRIPTION
**Pull request by tmjd**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: calico_node/tests/st/bgp/test_backends.py [hunk 1]
```diff
@@ -40,8 +43,20 @@ def test_bgp_backends(self):
...
+                    'kind': 'BGPConfiguration',
+                    'metadata': {
+                        'name': 'default',
+                    },
+                    'spec': {
+                        'ASNumber': LARGE_AS_NUM
+                    }
+                }
+
+            host1.writefile("bgpconfig.yaml", bgpconfig)
```
Comment (lwr20) [inline calico_node/tests/st/bgp/test_backends.py]: Use `update_bgp_config()` instead?

File: calico_node/tests/st/bgp/test_single_route_reflector.py [hunk 2]
```diff
@@ -41,8 +41,20 @@ def _test_single_route_reflector(self, backend='bird'):
...
+                        'name': 'default',
+                    },
+                    'spec': {
+                        'asNumber': 12345,
+                        'nodeToNodeMeshEnabled': False
+                    }
+                }
+
+            host1.writefile("bgpconfig.yaml", bgpconfig)
+            host1.calicoctl("apply -f bgpconfig.yaml")
```
Comment (lwr20) [inline calico_node/tests/st/bgp/test_single_route_reflector.py]: use `update_bgp_config()`?

File: calico_node/tests/st/bgp/test_global_peers.py [hunk 3]
```diff
@@ -51,7 +51,16 @@ def _test_global_peers(self, backend='bird'):
...
+            bgpconfig = {
+                    'apiVersion': 'projectcalico.org/v2',
+                    'kind': 'BGPConfiguration',
+                    'metadata': { 'name': 'default', },
+                    'spec': { 'nodeToNodeMeshEnabled': False }
+                }
+
+            host1.writefile("bgpconfig.yaml", bgpconfig)
+            # Set the default AS number.
+            host1.calicoctl("apply -f bgpconfig.yaml")
```
Comment (lwr20) [inline calico_node/tests/st/bgp/test_global_peers.py]: use `update_bgp_config()` instead?

File: calico_node/tests/st/bgp/test_ipip.py [hunk 4]
```diff
@@ -361,12 +371,18 @@ def _test_gce_int(self, with_ipip, backend, host1, host2, rrc):
...
-            host1.calicoctl("config set asNumber 64513")
-            host1.calicoctl("config set nodeToNodeMesh off")
+            bgpconfig = {
+                    'apiVersion': 'projectcalico.org/v2',
+                    'kind': 'BGPConfiguration',
+                    'metadata': { 'name': 'default', },
+                    'spec': { 'asNumber': 64513, 'nodeToNodeMeshEnabled': False} }
+
+            host1.writefile("bgpconfig.yaml", bgpconfig)
+            host1.calicoctl("apply -f bgpconfig.yaml")
```
Comment (lwr20) [inline calico_node/tests/st/bgp/test_ipip.py]: Use `update_bgp_config()` instead?

File: calico_node/tests/st/bgp/test_node_peers.py [hunk 5]
```diff
@@ -52,12 +53,20 @@ def _test_node_peers(self, backend='bird'):
...
-            host1.calicoctl("config set nodeToNodeMesh off")
+            bgpconfig = {
+                    'apiVersion': 'projectcalico.org/v2',
+                    'kind': 'BGPConfiguration',
+                    'metadata': { 'name': 'default',},
+                    'spec': { 'nodeToNodeMeshEnabled': False }}
+
+            host1.writefile("bgpconfig.yaml", bgpconfig)
+            # Set the default AS number.
+            host1.calicoctl("apply -f bgpconfig.yaml")
```
Comment (lwr20) [inline calico_node/tests/st/bgp/test_node_peers.py]: Use `update_bgp_config()` instead?

File: calico_node/tests/st/utils/utils.py [hunk 6]
```diff
@@ -223,6 +223,42 @@ def check_bird_status(host, expected):
                   "Output: \n%s" % (ipaddr, peertype, state, output)
             raise AssertionError(msg)
 
+@debug_failures
+def update_bgp_config(host, nodeMesh=None, asNum=None):
```
Comment (lwr20) [inline calico_node/tests/st/utils/utils.py]: Is it worth having a flag for choosing whether you set BGP config using a `BGPConfigurationList` or just a `BGPConfiguration`?

Comment (lwr20) [reply calico_node/tests/st/utils/utils.py]: Do we need to add tests where we check the function of multiple `BGPConfiguration`s in a list?

Comment (tmjd) [reply calico_node/tests/st/utils/utils.py]: I would think tests like that should go into calicoctl... though we do have calicoctl tests in here so maybe?

Comment (tmjd) [reply calico_node/tests/st/utils/utils.py]: I'd prefer that what Neil and I are working on in the STs right now though is just kept to the scope of getting the current tests working.  This already seems like a big enough task as it is.

**Pull request closed without merging**